xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -727,7 +727,7 @@ std::vector<CGUIFontTTF::Glyph> CGUIFontTTF::GetHarfBuzzShapedGlyphs(const vecTe
   std::vector<hb_script_t> scripts;
   std::vector<RunInfo> runs;
   hb_unicode_funcs_t* ufuncs = hb_unicode_funcs_get_default();
-  hb_script_t lastScript;
+  hb_script_t lastScript{HB_SCRIPT_INVALID};
   int lastScriptIndex = -1;
```

Comment:
lastScriptIndex seems to exist only to guard the non-initialized state of lastScript, so not needed anymore? (with slight alterations of the following code)
`if (lastScriptIndex != -1)` turns into `if (lastScriptIndex != HB_SCRIPT_INVALID)`
and remove
`lastScriptIndex = i;`

Reply:
Yes, seems similar case that other change in AEStreamInfo

Reply:
I'm not sure about this actually, is there a guarantee that the scripts vector will not contain HB_SCRIPT_INVALID values (ie hb_unicode_script() won't ever return HB_SCRIPT_INVALID)?
The harfbuzz man page is not very helpful and I am not familiar with the subject, it would be safer to ignore my previous comment.

Reply:
Did you think about this, whether you're sure HB_SCRIPT_INVALID won't be returned, or you rollback?

Reply:
I'm 99% sure it is only a placeholder for uninitialized value but I can't confirm 100%. Maybe worth merge as is now and  if some bad happens anyway seems easy to locate origin...

Reply:
Probably. harfbuzz is open source but I can't say I'm motivated to search the source code...

Reply:
Reverted the non clear change

Reply:
it's safer. why take a chance, this PR should be no-risk variable initialization.